extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/conf/ConfigurationService.java

```diff
@@ -68,4 +70,20 @@ public String getRedirectURI() throws GuacamoleException {
         return environment.getRequiredProperty(CASGuacamoleProperties.CAS_REDIRECT_URI);
     }
 
+    /**
+     * Returns the path to the file that contains the private key
+     * used to decrypt the credential that is sent encrypted by CAS,
+     * or null if no key is defined.
+     *
+     * @return
+     *     The path to the private key to decrypt the ClearPass
+     *     credential returned by CAS.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public Cipher getClearpassCipher() throws GuacamoleException {
+        return environment.getProperty(CASGuacamoleProperties.CAS_CLEARPASS_KEY);
+    }
+
```

Comment:
Taking a key argument and calling this a cipher property feels a little kludgy.  Kind of goes along with the question about how far the Cipher property should go or if that's even really a good name/function for it.

Reply:
I would tend to agree there. Why not `PrivateKey`?

Reply:
Yeah, PrivateKey is probably a better route to go, and just keep the Cipher pieces of it in the CAS extension code.  I'll try to rework this a little bit.

Reply:
Refactored as PrivateKey.